cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -44,66 +48,54 @@
-                            .write(data);
-                    request.result().send(response -> {
-                        if (response.succeeded()) {
-                            if (response.result().statusCode() == 200 || response.result().statusCode() == 201) {
-                                response.result().bodyHandler(buffer -> {
-                                    try {
-                                        @SuppressWarnings({ "rawtypes" })
-                                        Map t = mapper.readValue(buffer.getBytes(), Map.class);
-                                        LOGGER.debugCr(reconciliation, "Got {} response to PUT request to {}: {}", response.result().statusCode(), path, t);
-                                        result.complete(t);
-                                    } catch (IOException e) {
-                                        result.fail(new ConnectRestException(response.result(), "Could not deserialize response: " + e));
-                                    }
-                                });
-                            } else {
-                                // TODO Handle 409 (Conflict) indicating a rebalance in progress
-                                LOGGER.debugCr(reconciliation, "Got {} response to PUT request to {}", response.result().statusCode(), path);
-                                response.result().bodyHandler(buffer -> {
-                                    result.fail(new ConnectRestException(response.result(), tryToExtractErrorMessage(reconciliation, buffer)));
-                                });
-                            }
-                        } else {
-                            result.tryFail(response.cause());
-                        }
-                    });
-                } else {
-                    result.fail(request.cause());
-                }
-            }));
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(String.format("http://%s:%d%s", host, port, encodeURLString(path))))
+                .PUT(HttpRequest.BodyPublishers.ofString(data))
+                .setHeader("Accept", "application/json")
+                .setHeader("Content-Type", "application/json")
+                .build();
+
+        return httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                .thenCompose(response -> {
+                    int statusCode = response.statusCode();
+                    if (statusCode == 200 || statusCode == 201) {
+                        JsonNode json = parseToJsonNode(response);
+                        Map<String, Object> t = OBJECT_MAPPER.convertValue(json, TREE_TYPE);
+                        LOGGER.debugCr(reconciliation, "Got {} response to PUT request to {}: {}", statusCode, path, t);
+                        return CompletableFuture.completedFuture(t);
+                    } else {
+                        // TODO Handle 409 (Conflict) indicating a rebalance in progress
```

Comment:
I know this was here before this PR. Do you think it is still an issue? If you don't know / or if you think it still needs to be looked into ... could you please open a GitHub issue for it and add a link to it to this comment? Thanks.

Reply:
Thanks so much for the review Jakub!

I'm not sure if this is still an issue or not so I have raised a Git issue for investigating it https://github.com/strimzi/strimzi-kafka-operator/issues/10933 and add its link to the comments.